src/v/coproc/event_listener.cc

```diff
@@ -80,17 +88,69 @@ event_listener::persist_actions(absl::btree_map<script_id, iobuf> wsas) {
 }
 
 event_listener::event_listener(ss::sharded<pacemaker>& pacemaker)
-  : _client({config::shard_local_cfg().kafka_api()[0].address})
-  , _dispatcher(pacemaker, _abort_source) {}
+  : _dispatcher(pacemaker, _abort_source) {}
+
+ss::future<> event_listener::do_connect() {
+    vassert(!_client, "do_connect() called when theres already a valid client");
+    /// Invoke metadata request on a new client connected to this broker
+    auto client = std::make_unique<kafka::client::transport>(
+      rpc::base_transport::configuration{
+        .server_addr = ss::socket_address(
+          ss::net::ipv4_address("127.0.0.1"), 9092)});
+    co_await client->connect();
```

Comment:
Connect will throw and hit my error handler, the loop will continue and retry again as _client is still nullptr. Could the connection still leak if connect throws?

Reply:
in one of the `if` statements below there is `co_return` without close

Reply:
Thats because we set _client to client and take the connection. But I think your still right in another case I throw partition_error, I don't call close